### PR TITLE
Add Tree concurrency tests

### PR DESCRIPTION
**Pull request by 7hong13**
#### What are the relevant tickets?
- https://github.com/yorkie-team/yorkie-js-sdk/pull/792

Fixes #

### Checklist

- [x] Added relevant tests or not required
- [x] Didn't break anything

## Summary by CodeRabbit

- **New Features**
  - Enhanced JSON tree editing capabilities with new methods.

- **Bug Fixes**
  - Updated OkHttpClient protocol to improve connection stability.
  - Improved document status management for better synchronization.

- **Improvements**
  - Upgraded `androidxLifecycle` and `androidx-annotation` dependencies for enhanced performance and stability.
  - Refactored test functions for improved readability and maintenance.
  - Consolidated internal test annotations for streamlined testing.

- **Chores**
  - Updated CI workflow configuration to use specific test classes.

### CODE REVIEW

**Comment by 7hong13** [comment]
Tests are failing because some server responses are missing, but it only happens on the local server. The tests work fine on the real server. I'll delve deeper into the failing tests to figure out what's causing this later.

File: yorkie/src/androidTest/kotlin/dev/yorkie/core/TestUtils.kt [hunk 1]
```diff
@@ -12,7 +12,7 @@ const val GENERAL_TIMEOUT = 3_000L
 fun createClient() = Client(
     "http://10.0.2.2:8080",
     unaryClient = OkHttpClient.Builder()
-        .protocols(listOf(Protocol.HTTP_1_1))
+        .protocols(listOf(Protocol.H2_PRIOR_KNOWLEDGE))
```
Comment (skhugh) [inline yorkie/src/androidTest/kotlin/dev/yorkie/core/TestUtils.kt]: we should stay on http 1 as http 2 is very unstable

File: yorkie/src/main/kotlin/dev/yorkie/document/Document.kt [hunk 2]
```diff
@@ -451,6 +451,10 @@ public class Document(
         onlineClients.value -= actorID
     }
 
+    internal fun setStatus(newStatus: DocumentStatus) {
```
Comment (skhugh) [inline yorkie/src/main/kotlin/dev/yorkie/document/Document.kt:454]: do we need additional setter method for this?
Kotlin's properties are different from Java's fields, and it is not wrong to use properties for simple getter/setter's.
(https://blog.kotlin-academy.com/kotlin-should-i-define-function-or-property-6786951da909)

File: yorkie/src/androidTest/kotlin/dev/yorkie/core/TestUtils.kt (hunk 1 above)
Comment (7hong13) [reply yorkie/src/androidTest/kotlin/dev/yorkie/core/TestUtils.kt]: using http2 is outdated code. I just tested it temporarily.

File: yorkie/src/main/kotlin/dev/yorkie/document/Document.kt (hunk 2 above)
Comment (7hong13) [reply yorkie/src/main/kotlin/dev/yorkie/document/Document.kt]: I thought setting the `Document.status` from `Client` directly can violate the encapsulation. I reverted it based on your suggestion: [8febcf6](https://github.com/yorkie-team/yorkie-android-sdk/pull/186/commits/8febcf6c7fdf75aa47a009bf59db7f22a756bc62)

**Pull request merged**